components/logins/android/src/main/java/mozilla/appservices/logins/ImportMetrics.kt

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.logins
+import org.json.JSONArray
+// import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Raw password data that is stored by the LoginsStorage implementation.
+ */
+data class ImportMetrics(
+
+    /**
+     * Total duration of the import in ms
+     */
+   val totalDuration: Int,
+
+    /**
+     * List of metrics for each import phase
+     */
+    val phases: List<ImportPhaseMetrics>,
+
+    /**
+     * Number of records that failed during the import
+     */
+    val numFailed: Int,
+
+    /**
+     * List of errors encountered during the import
+     */
+    val errors: List<String>
+) {
+
+    private fun phasesToJSON(phaseList: List<ImportPhaseMetrics>): JSONArray {
+        val phaseArray = JSONArray()
+        for (item in phasesList) {
+            phaseArray.put(item.toJSON())
+        }
+        return phaseArray
+    }
+
+    fun toJSON(): JSONObject {
```

Comment:
I don't quite understand how we are going to get this into glean yet, but it seems extravagant to have both toJSON and fromJSON, and if they are both necessary it begs the question of whether we can avoid this entirely and just pass around the json string we got from rust. I might be naive, but it's worth keeping in mind if there's any possibility of doing this and removing this file entirely.

:wishfulthinking: :)

Reply:
I can't think of a reason why application code would need to convert one of these back into JSON (and certainly we can't use that for getting the data into glean). But perhaps it will be useful for tests or similar.

Reply:
I honestly wasn't sure about the necessity of the `ImportMetrics` and `ImportPhaseMetrics` classes. Maybe it's enough to return `JSONObject` in the Kotlin import logins functions? I'm just not sure what will work best for the caller once the returned value is received and submitted to Glean.

Reply:
I honestly wasn't sure about the necessity of the `ImportMetrics` and `ImportPhaseMetrics` classes. Maybe it's enough to return `JSONObject` in the Kotlin import logins functions? I'm just not sure what will work best for the caller once the returned value is received and submitted to Glean.